pkg/infrastructure/openstack/clusterapi/clusterapi.go

```diff
@@ -71,6 +74,30 @@ func (p Provider) PreProvision(ctx context.Context, in clusterapi.PreProvisionIn
 		}
 	}
 
+	{
+		capiMachines := make([]capo.OpenStackMachine, 0, len(machineManifests))
+		for i := range machineManifests {
+			if m, ok := machineManifests[i].(*capo.OpenStackMachine); ok {
+				capiMachines = append(capiMachines, *m)
+			}
+		}
+
+		workerSpecs := make([]mapov1alpha1.OpenstackProviderSpec, 0, len(machineManifests))
```

Comment:
should we use `len(workers)` instead of `len(machineManifests)`?

Reply:
uh-oh another good catch, that's a copy-paste error.